core/optionsx.h

```diff
@@ -1158,9 +1159,11 @@ OPTION_DEFAULT(uint, max_trace_bbs, 128, "maximum number of basic blocks in a tr
 /* FIXME i#3522: re-enable SELFPROT_DATA_RARE on linux */
 OPTION_DEFAULT(
     uint, protect_mask,
-    IF_STATIC_LIBRARY_ELSE(0x100 /*SELFPROT_GENCODE*/,
-                           IF_WINDOWS_ELSE(0x101 /*SELFPROT_DATA_RARE|SELFPROT_GENCODE*/,
-                                           0x100 /*SELFPROT_GENCODE*/)),
+    IF_STATIC_LIBRARY_ELSE(
+        0x100 /*SELFPROT_GENCODE*/,
+        IF_MACOSA64_ELSE(0,
```

Comment:
There's really no support for changing +rwx memory to +rx?  I thought that's what the jit_write_* calls you added did?

Reply:
Added an XXX comment.